kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+
+FromJSONCheckResultProcess::FromJSONCheckResultProcess(
+    Model& rModel,
+    Parameters ThisParameters
+    ) : mrModelPart(GetModelPartFromModelAndSettings(rModel, ThisParameters)),
+        mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::FromJSONCheckResultProcess(
+    ModelPart& rModelPart,
+    Parameters ThisParameters
+    ) : mrModelPart(rModelPart),
+        mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::Execute()
+{
+    KRATOS_TRY;
+
+    ExecuteInitialize();
+    ExecuteFinalizeSolutionStep();
+    ExecuteFinalize();
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::ExecuteInitialize()
+{
+    KRATOS_TRY;
+
+    // The auxiliar vectors for the variables
+    const auto& r_check_variables_names = mThisParameters["check_variables"].GetStringArray();
+    const auto& r_gauss_points_check_variables = mThisParameters["gauss_points_check_variables"].GetStringArray();
+
+    // We fill the list of variables
+    FillVariablesList(r_check_variables_names, r_gauss_points_check_variables);
+
+    // Set some flags
+    this->Set(HISTORICAL_VALUE, mThisParameters["historical_value"].GetBool());
+    this->Set(USE_NODE_COORDINATES, mThisParameters["use_node_coordinates"].GetBool());
+    KRATOS_WARNING_IF("FromJSONCheckResultProcess", this->Is(USE_NODE_COORDINATES)) << "use_node_coordinates is deprecated. Please do not use it" << std::endl;
+    this->Set(CHECK_ONLY_LOCAL_ENTITIES, mThisParameters["check_only_local_entities"].GetBool());
+
+    // We initialize the databases
+    InitializeDatabases();
+
+    // Additional values
+    mFrequency = mThisParameters["time_frequency"].GetDouble();
+    mAbsoluteTolerance = mThisParameters["tolerance"].GetDouble();
+    mRelativeTolerance = mThisParameters["relative_tolerance"].GetDouble();
+    const std::size_t absolute_tolerance_digits = ComputeRelevantDigits(mAbsoluteTolerance);
+    const std::size_t relative_tolerance_digits = ComputeRelevantDigits(mRelativeTolerance);
+    mRelevantDigits = static_cast<std::size_t>(std::max(absolute_tolerance_digits, relative_tolerance_digits)) + 1;
+
+    // Set the flag CORRECT_RESULT to false
+    this->Set(CORRECT_RESULT, false);
+
+#ifdef KRATOS_DEBUG
+    // We do the check
+    Check();
+#endif
```

Comment:
Do you think it is possible to do it also in `Release` mode? (just asking)

Reply:
It isquite expensive check, I prefer not